supplier/tests/test_supplier_page.py

```diff
@@ -32,10 +78,57 @@ def test_fail_entry(client, non_supplier_mock):
 @pytest.mark.django_db()
 def test_supplier_check_false(non_supplier_mock):
     assert non_supplier_mock not in Supplier.objects.all()
-    assert not supplier_check(non_supplier_mock.supplier_account)
+    assert not get_supplier(non_supplier_mock.supplier_account)
 
 
 @pytest.mark.django_db()
 def test_supplier_check_true(saved_supplier0):
     assert saved_supplier0 in Supplier.objects.all()
-    assert supplier_check(saved_supplier0) == saved_supplier0
+    assert get_supplier(saved_supplier0) == saved_supplier0
+
+
+@pytest.mark.django_db()
+def test_receive_product_return_supplier_product(request_searched_product_mock,
+                                                 saved_supplier0,
+                                                 saved_supplier_product0,
+                                                 saved_supplier_product1,
+                                                 searched_product_string):
+    assert saved_supplier_product0.user_name == saved_supplier0
+    assert saved_supplier_product1.user_name == saved_supplier0
+    request_searched_product_mock.GET[searched_product_string] = saved_supplier_product0.qr_code.product_name
+    assert receive_product(request_searched_product_mock, saved_supplier0).first() == saved_supplier_product0
+
+
+@pytest.mark.django_db()
+def test_receive_product_return_none(request_searched_product_mock,
+                                     saved_supplier0,
+                                     searched_product_string):
+
+    request_searched_product_mock.GET[searched_product_string] = 'Fizzibublach'
+    assert not receive_product(request_searched_product_mock, saved_supplier0)
+
+
+@pytest.mark.django_db()
+def test_display_all_products(get_request,
+                              render_mock,
+                              all_products_mock,
+                              saved_supplier0):
+    data = {'supplier_products': all_products_mock.return_value}
+    render_mock.return_value = render(get_request, 'supplier/suppliers.html', data)
+    assert action_display_all_products(get_request, saved_supplier0) == render_mock.return_value
+
+
+@pytest.mark.django_db()
+def test_searched_product_with_data(get_request_with_data,
+                                    render_mock,
+                                    saved_supplier0,
+                                    get_request,
+                                    saved_supplier_product0):
+    # test = action_searched_product(get_request_with_data, saved_supplier0)
```

Comment:
I guess you are still working on this. Please add `[WIP]` to the title of this PR.